app/src/main/java/org/wikipedia/search/SearchResultsFragment.kt

```diff
@@ -0,0 +1,537 @@
+                    { searchResponse: PrefixSearchResponse?, readingListSearchResults: SearchResults?, historySearchResults: SearchResults ->
+                        val searchResults: SearchResults = if (searchResponse?.query()!!.pages() != null) {
+                            // noinspection ConstantConditions
+                            SearchResults(searchResponse.query()!!.pages()!!,
+                                    WikiSite.forLanguageCode(searchLanguageCode!!), searchResponse.continuation(),
+                                    searchResponse.suggestion())
+                        } else {
+                            // A prefix search query with no results will return the following:
+                            //
+                            // {
+                            //   "batchcomplete": true,
+                            //   "query": {
+                            //      "search": []
+                            //   }
+                            // }
+                            //
+                            // Just return an empty SearchResults() in this case.
+                            SearchResults()
+                        }
+                        handleSuggestion(searchResults.suggestion)
+                        val resultList: MutableList<SearchResult> = ArrayList()
+                        addSearchResultsFromTabs(resultList)
+                        resultList.addAll(readingListSearchResults!!.results)
+                        resultList.addAll(historySearchResults.results)
+                        resultList.addAll(searchResults.results)
+                        resultList
+                    })
+        }
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doAfterTerminate { updateProgressBar(false) }
+                .subscribe({ results: MutableList<SearchResult> ->
+                    binding.searchErrorView.visibility = View.GONE
+                    handleResults(results, searchTerm!!, startTime)
+                }) { caught: Throwable? ->
+                    binding.searchErrorView.visibility = View.VISIBLE
+                    binding.searchErrorView.setError(caught)
+                    binding.searchResultsContainer.visibility = View.GONE
+                    logError(false, startTime)
+                })
+    }
+
+    private fun addSearchResultsFromTabs(resultList: MutableList<SearchResult>) {
+        if (currentSearchTerm!!.length < 2) {
+            return
+        }
+        val tabList: List<Tab> = WikipediaApp.getInstance().tabList
+        for (tab in tabList) {
+            if (tab.backStackPositionTitle != null &&
+                    tab.backStackPositionTitle!!.displayText.toLowerCase(Locale.getDefault())
+                            .contains(currentSearchTerm!!.toLowerCase(Locale.getDefault()))) {
```

Comment:
Should we use `Locale.getRoot` instead?

Reply:
Why do you think Root is more appropriate here?

Reply:
I don't see the difference of the output between `Locale.ROOT` vs `Locale.getDefault()`, but I do notice we can change the default locale if we want, and also the IDE suggestion for the `.toLowerCase()` is to set `Locale.ROOT`.

Reply:
When we were dealing with the date issue on watchlist item string, the ROOT was not playing nice.. I had to finally use .getDefault... but going by this documentation the diff is ROOT could be empty for Locale but getDefault will have the Locale of the JVM's instance [ assuming always non null or empty] ... still do not fully understand the diff..

Reply:
https://developer.android.com/reference/java/util/Locale

Reply:
I see. It makes sense to me. Let's keep `getDefault()` then.